ci/playbooks/multinode-customizations.yml

```diff
@@ -9,85 +9,135 @@
-          log-queries
-          no-negcache
-          local=/crc.testing/
-          domain=crc.testing
-          address=/apps-crc.testing/{{ _crc_default_net_ip }}
-          address=/api.crc.testing/{{ _crc_default_net_ip }}
-          address=/api-int.crc.testing/{{ _crc_default_net_ip }}
-          address=/crc-74q6p-master-0.crc.testing/{{ _crc_default_net_ip }}
-          {% for dns_server in _crc_default_net_dns -%}
-          server={{ dns_server }}
-          {% endfor %}
-          EOL
-
-          cat << EOL | sudo tee /etc/resolv.conf
-          nameserver {{ _crc_default_net_ip }}
-          EOL
-
-          # stop overwriting /etc/resolv.conf after reboot
-          cat << EOL | sudo tee /etc/NetworkManager/conf.d/00-custom-crc.conf
-          [main]
-          dns=none
-          EOL
-
-          sudo systemctl reload NetworkManager
-
-    - name: Restart the service because zuul base job is overwriting /etc/resolv.conf
-      become: true
-      ansible.builtin.systemd:
-        state: restarted
-        name: crc-pre
+      block:
+        - name: Manage name resolution and interfaces
+          become: true
+          vars:
+            _crc_default_iface: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.iface
+              }}
+            _crc_default_gw: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.gw
+              }}
+            _crc_private_connection_name: >-
+              {{
+                 _decoded_net_env.crc_ci_bootstrap_networks_out.crc.default.connection
+              }}
+          block:
+            - name: Ensure crc knows about its second NIC
+              community.general.nmcli:
+                autoconnect: true
+                conn_name: "{{ _crc_private_connection_name }}"
+                dns4: 127.0.0.1
+                ifname: "{{ _crc_default_iface }}"
+                type: ethernet
+                ip4: "{{ _crc_default_net_ip }}"
+                gw4: "{{ _crc_default_gw }}"
+                state: present
 
-    # Avoid 'state: restarted' due to issues with IP not available when crc-dnsmasq starts
-    - name: Stop dnsmasq
-      become: true
-      ansible.builtin.systemd:
-        state: stopped
-        name: crc-dnsmasq
+            - name: Ensure crc does not get "public" DNS
+              become: true
+              community.general.nmcli:
+                autoconnect: true
+                conn_name: "Wired connection 1"
+                dns4_ignore_auto: true
+                state: present
+
+            - name: Restart NetworkManager
+              ansible.builtin.service:
+                name: NetworkManager
+                state: reloaded
+
+        - name: Configure dnsmasq
+          vars:
+            _dnsmasq_config: >-
+              {{
+                _dnsmasq.stat.exists |
+                ternary('/srv/dnsmasq.conf', '/etc/dnsmasq.d/crc-dnsmasq.conf')
```

Comment:
added an earlier comment perhaps use _dnsmasq.stat.path as the true result instead of repeating the path.

do we *also* want to stat /etc/dnsmasq.d/crc-dnsmasq.conf as an extra sanity then just error if neither?

Reply:
ah, right for the .path - missed that one while doing the deeper update.

As for checking the other file: it would fail if it doesn't exist anyway, because we don't instruct ansible to `create: true` the file.
AFAIK, the next move CRC *could* do is moving dnsmasq to the NetworkManager service... Not sure they want to go that way though.